repository.go

```diff
@@ -990,7 +990,75 @@ func (r repository) preload(cw contextWrapper, records slice, field string, quer
 	scanFinish := r.instrumenter.Observe(cw.ctx, "rel-scan-multi", "scanning all records to multiple targets")
 	defer scanFinish(nil)
 
-	return scanMulti(cur, keyField, keyType, targets)
+	err = scanMulti(cur, keyField, keyType, targets)
+	if err != nil {
+		return err
+	}
+
+	return cleanUpAssociationFields(records, field, path)
```

Comment:
I think introducing additional loop here is not the best way to do this

maybe we can try adding lazy version of Assoication.Document and Association.Collection that returns slice interface. where in the lazy version initialization of doc and collection will happens only if Add() is called 🤔

Reply:
Implemented `Association.LazyDocument`.